assets/performanceprofile/scripts/set-rps-mask.sh

```diff
@@ -1,16 +1,37 @@
 #!/usr/bin/env bash
 
-path=${1}
-[ -n "${path}" ] || { echo "The device path argument is missing" >&2 ; exit 1; }
-
-mask=${2}
-[ -n "${mask}" ] || { echo "The mask argument is missing" >&2 ; exit 1; }
-
+function set_queue_rps_mask() {
 queue_path=${path%rx*}
 
 queue_num=${path#*queues/}
+
 # replace '/' with '-'
 queue_num="${queue_num/\//-}"
 
 # set rps affinity for the queue
-echo "${mask}" > "/sys${queue_path}${queue_num}/rps_cpus"
\ No newline at end of file
+echo "${mask}"  2> /dev/null > "/sys${queue_path}${queue_num}/rps_cpus"
+
+# the 'echo' command might failed if the device path which the queue belongs to has changes
+# this can happen in case of SRI-OV devices renaming
+exit 0
```

Comment:
The idea is ok but forcing an exit 0 will make the function non-composable. The calling program will always exit here, won't it? If we want to swallow the error from echo, I'd either use `return 0` or `echo "${mask}" 2> /dev/null > "/sys${queue_path}${queue_num}/rps_cpus" || :`

Reply:
We should return 0 in the main flow context (at the end of the file) instead of the function I think

Reply:
We don't want to exit 0 in the main flow because we still want to catch errors if the second function is being called

Reply:
The `set_net_dev_rps` one

Reply:
ok, we do want swallow error of `echo` on line 12 though, right?

Reply:
Yes and you're right we should return 0 and not exit. it would basically have the same affect but will keep the function composable